Molyktech/Property-Pro-Lite
# #166871320 Build out end point for user to signup

#### What does this PR do?
add  endpoint to  signup a new user

#### Description of Task to be completed?
have the  endpoint working:
`` POST  'http://localhost:3000/api/v1/auth/signup``
validate incoming request with joi
hash user password using bcrypt
respond with status and newly created user

#### How should this be manually tested?
``clone the repo``
``cd into the directory Property-Pro-Lite``
``run npm install``
``run npm start``
``using POSTMAN, test by making a POST request to 'http://localhost:3000/api/v1/auth/signup``

make sure the header is set to:
 `` key: Content-Type ``
``value: application/json ``


#### Any background context you want to provide?
NIL

#### What are the relevant pivotal tracker stories?
[#166871320](https://www.pivotaltracker.com/story/show/166871320)

#### Screenshots (if appropriate)
NIL

#### Questions:
NIL
